### DISCUSSION TITLE
Is it possible to change port of LocalAI to run on Jupyter Lab on Vertex AI?

### DISCUSSION THREAD

**Discussion by bobbyjacob** [mudler/LocalAI#292] [Q&A] [answered]
Something is already holding 8080 on Vertex AI workbench. How do I change the port so I can run it on Vertex AI?

```
docker compose up
[+] Running 2/1
 ✔ Network localai_default  Created                                                                                          0.1s 
 ✔ Container localai-api-1  Created                                                                                          0.1s 
Attaching to localai-api-1
Error response from daemon: driver failed programming external connectivity on endpoint localai-api-1 (6ccc3cd9ee58044974f7acebdd22879c202d5fc0d48ed05de5eb5b44f47ef0af): Error starting userland proxy: listen tcp4 0.0.0.0:8080: bind: address already in use
(base) jupyter@gen-ai-demo:~/LocalAI/examples/langchain-chroma$ 
```

**Comment by mudler** [comment, marked as answer]
You can change the port over here: https://github.com/go-skynet/LocalAI/blob/aa7a18f131547537ad0df3808aec895e0e438c22/docker-compose.yaml#L10

**Comment by bobbyjacob** [comment]
This worked. Thanks! Specifically in `LocalAI/examples/langchain-chroma/docker-compose.yml` I changed the `ports` line to this:
```
    ports:
      - 8081:8080
```
This pointed the host port (8081) to point the the container port that localai was running on (8080).

**Reply by mudler** [reply]
awesome, glad it works now!

**Comment by lmaddox** [comment]
"If running in p2p mode with container images, make sure you start the container with --net host or network_mode: host in the docker-compose file." [source](https://localai.io/features/distribute/#notes)

clearly there must be some under-documented flag or env var to tell the localai server to listen on another port, since a port mapping won't work with the network mode in the docs.
